tools/autograd/derivatives.yaml

```diff
@@ -277,19 +306,22 @@
 - name: bmm(Tensor self, Tensor mat2) -> Tensor
   self: grad.bmm(mat2.transpose(1, 2).conj())
   mat2: self.transpose(1, 2).conj().bmm(grad)
+  result: bmm_forward(self_fw_grad, mat2_fw_grad, self, mat2)
```

Comment:
It seems you have some magic convention whereby every input argument `x` gets a `x_fw_grad` implicitly defined. First, if you're going to do this, you need to document the convention. But second, you don't need a magic convention; forward gradients are attached to the input tensors, so you can literally just write `self.tangent()` and now you don't need a magic convention. (Of course, feel welcome to match for .tangent() when processing expressions and CSE repeated uses.)